udmis/etc/k8s_etcd.yaml

```diff
@@ -0,0 +1,92 @@
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: etcd-ssd
+provisioner: kubernetes.io/gce-pd
+parameters:
+  type: pd-ssd
+  zone: us-central1-f
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: etcd-client
+spec:
+  type: LoadBalancer
+  ports:
+  - name: etcd-client
+    port: 2379
+    protocol: TCP
+    targetPort: 2379
+  selector:
+    app: etcd-set
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: etcd-set
+spec:
+  type: ClusterIP
+  clusterIP: None
+  ports:
+  - name: client
+    port: 2379
+    protocol: TCP
+    targetPort: 2379
+  - port: 2380
+    name: peer
+  selector:
+    app: etcd-set
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: etcd-set
+  labels:
+    app: etcd-set
+spec:
+  serviceName: etcd-set
+  replicas: 3
+  selector: 
+    matchLabels:
+      app: etcd-set
+  template:
+    metadata:
+      name: etcd-set
+      labels:
+        app: etcd-set
+    spec:
+      containers:
+      - name: etcd-set
+        image: quay.io/coreos/etcd:latest
```

Comment:
consider pinning this to a specific version, such as the one set above. In general, for security and compatibility reasons it is recommended to pin container images to a specific version. Observability tools and such can provide information that the container is out of date.

For the CoreOS etcd image it may not be a big deal, but if you were to say, pull a container image from a random outsider, they could attack any infra by pushing new versions to latest.

Reply:
sure -- it's what I usually do, this was just cut-and-paste stuff I wasn't paying attention to...  Usually I find it's not security or compatibility reasons, but rather stability and consistency -- without a pinned version things can randomly change and randomly introduce bugs that are then really confusing as to why things suddenly started to behave differently.

Reply:
Well then. It turns out that the "latest" tag is actually 6 years old, and that the more recent versions don't have a shell in them... it's possible, but would require rejiggering this a bit to get the initial etcd startup to work -- not something I would like to tackle just now.  So, gonna put it on "the list" of things to do. Likely need to setup a config-map for the cmdline parameters is all.  This of course will all come back to haunt me horribly when they do change the "latest" tag for some reason and everything suddenly decides to break...